src/components/StatsPanel/StatsPanel.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { useStatsPanel } from './StatsPanel.hook';
+import {
+  CardContainer,
+  StatItem,
+  StatIcon,
+  StatText,
+  StatValue,
+  IconWrapper,
+} from './StatsPanel.styles';
+
+import type { StatsPanelProps } from './StatsPanel.types';
+
+export const StatsPanel = ({ totalTests, passed, failed, waiting }: StatsPanelProps) => {
+  const stats = useStatsPanel({ totalTests, passed, failed, waiting });
+
+  return (
+    <CardContainer>
+      {stats.map((stat) => {
+        const Icon = stat.icon;
+
+        return (
+          <StatItem key={stat.id}>
+            <IconWrapper color={stat.color}>
+              <StatIcon color={stat.color}>
+                {typeof Icon === 'string' ? (
```

Comment:
When Icon is a string the component renders an <img src={Icon} />. External SVGs loaded via <img> will not inherit the CSS fill/stroke or color from the surrounding styles and therefore cannot be recolored by theme. This contradicts the styles which attempt to color icons via CSS.

**Sugerencia:**
```typescript
Prefer importing SVGs as React components (SVGR) so they are inline DOM nodes and can inherit currentColor. Example: import { ReactComponent as LikeIcon } from './assets/icons/like.svg' and then render <LikeIcon aria-hidden />. Update your svg typings to support ReactComponent (see suggestion for custom.d.ts).
```